shim/src/hashtable.c

```diff
@@ -51,13 +51,12 @@ int hashtable_insert(struct hashtable *h, int key, uint64_t val)
     int skip = 0;
     int hash = 0;
     const int length = (1 << h->length_log2);
-    const int mask = (length - 1);
 
     assert(h != NULL);
     assert(key != HASHTABLE_NULL);
     assert(val != (uint64_t)HASHTABLE_NULL);
 
-    hash = key & mask;
+    hash = key % length;
```

Comment:
Since length is a power of two, why should we use expensive modulo operations?

Reply:
Ok, this makes sense. I didn't realize that the modulo operation was more expensive. I reverted it back to the mask.